addons/html_editor/static/src/main/chatgpt/chatgpt_plugin.js

```diff
@@ -82,7 +82,56 @@ export class ChatGPTPlugin extends Plugin {
         return cannotReplace || isEmpty;
     }
 
-    openDialog(params = {}) {
+    /**
+     * Converts record data to JSON, so we can pass them to the AI record's context
+     * @returns {String} String JSON representation of the record
+     */
+    recordDataToJSON(recordData, fieldsInfo) {
```

Comment:
With a name like that it's surprising to see so much lines of code because you'd expect a simple call to JSON and be done with it, barring a few special cases. My point is that the function does more than simply converting between data formats. I suggest renaming it to something akin to `recordDataToContextJSON`.